From 85de34c9bc44a38104a32a10b7aa615fbe147664 Mon Sep 17 00:00:00 2001
From: Simon McVittie <simon.mcvittie@collabora.co.uk>
Date: Thu, 27 Oct 2016 15:06:01 +0100
Subject: [PATCH 3/5] Add 'needs-reboot' restriction

This allows tests to be explicit about needing to reboot the machine,
rather than assuming that 'isolation-machine' is enough. It is plumbed
into the existing 'reboot' capability, which is distinct from
'isolation-machine'.

Signed-off-by: Simon McVittie <simon.mcvittie@collabora.co.uk>
---
 doc/README.package-tests.rst | 4 ++++
 lib/testdesc.py              | 8 +++++++-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/doc/README.package-tests.rst b/doc/README.package-tests.rst
index 3eb1230..4775a07 100644
--- a/doc/README.package-tests.rst
+++ b/doc/README.package-tests.rst
@@ -202,6 +202,10 @@ isolation-machine
     test in a virtualization server which does not provide this it will
     be skipped.
 
+needs-reboot
+    The test wants to reboot the machine using
+    ``/tmp/autopkgtest-reboot`` (see below).
+
 needs-recommends
     Enable installation of recommended packages in apt for the test
     dependencies. This does not affect build dependencies.
diff --git a/lib/testdesc.py b/lib/testdesc.py
index a8eca66..914e00c 100644
--- a/lib/testdesc.py
+++ b/lib/testdesc.py
@@ -42,7 +42,7 @@ import adtlog
 
 known_restrictions = ['rw-build-tree', 'breaks-testbed', 'needs-root',
                       'build-needed', 'allow-stderr', 'isolation-container',
-                      'isolation-machine', 'needs-recommends']
+                      'isolation-machine', 'needs-recommends', 'needs-reboot']
 
 
 class Unsupported(Exception):
@@ -161,6 +161,12 @@ class Test:
                               'Test needs root on testbed which is not '
                               'available')
 
+        if 'needs-reboot' in self.restrictions and \
+           'reboot' not in caps:
+            raise Unsupported(self.name,
+                              'Test needs to reboot testbed but testbed does '
+                              'not provide reboot capability')
+
 #
 # Parsing for Debian source packages
 #
-- 
2.10.1

